158595336

Darius-Ndubi opened the pull request:
## What does this PR do?

- User sign in endpoint to verify entered credentials.
- Adds un-signed up user rejection to sign in endpoint.

## What is the relevant PT story?
[#158595336](https://www.pivotaltracker.com/story/show/158595336)

Darius-Ndubi committed:
[Feature #158595336]
User signin endpoint
Adds un-signed up user rejection

esirK commented:
Kindly push your other fixes so that I can review them.

Darius-Ndubi commented:
okay let me do that just now

Darius-Ndubi committed:
[Fix #158608973] Removes inner loops in ride info retrieval
Renames variables that start with capital letters
Reconstructs ride variables in data.py to match with those in views.py

Darius-Ndubi committed:
Removes all bin files pushed earlier

esirK commented on api/app/views.py:
```diff
@@ -10,19 +10,16 @@
         #add the id and title to dictionary
-        All_rides.update({ride_id: ride_title})
-    return jsonify(All_rides)
+        all_rides.update({ride_id: ride_title})
```
according to the name `all_rides`, would someone expect to find rides with full details of just the ride title??

esirK commented on api/app/views.py:
```diff
@@ -100,15 +86,13 @@ def ride_request(id, requests):
+        if ride.get('id')==id:
+            to_delete = ride
+            #find the index of the ride
+            ride_index = rides.index(to_delete)
```
are you sure this will work??

esirK commented on api/app/views.py:
```diff
@@ -119,40 +103,36 @@ def edit_ride(id):
-                    return jsonify({"You want to edit":to_edit})
+            if ride.get('id') == id:
+                to_edit=ride
+                return jsonify({"You want to edit":to_edit})
```
is this necessary??

Darius-Ndubi committed:
[Chore #158615644]

Makes tests Post method work
Adds mock data to data.py
Edits data properties in views.py that were miss typed

Darius-Ndubi committed:
Adds api documentation

Darius-Ndubi committed:
Redesigns the API to use schemas

esirK commented on api/apidoc/app.py:
```diff
@@ -1,224 +1,179 @@
+        schema=UsersSchema()
+        result=schema.dumps(user)
+        print result
+        users.append(result)
```
we agreed you should add a `user` object and not the `json` representation of the user object right??

Darius-Ndubi committed:
Modifies the whole API documentation as recommended by LFA

esirK commented on api/apidoc/models.py:
```diff
@@ -0,0 +1,56 @@
+        
+
+    
+    def addRide(self,ride_data):
```
Who is responsible for managing rides??

esirK commented on api/apidoc/data.py:
```diff
@@ -0,0 +1,101 @@
+rides = [
+    {
```
Why not just leave this as an empty list??

wandesky reviewed (commented):
I wrote this to illustrate to @Darius-Ndubi how comments are done in github

Darius-Ndubi committed:
[Chore  #158615644] [Bug #158690334] [Fix #158690070] [Bug #158689970] [Fix #158689919] [Fix #158690202]

Removes mock data from data.py
Fixes bugs in edit ride api documentation endpoint
Fixes bug in sigin api documentation endpoint
Fixes endpoits to mimic realsites actions
Fixes display oprions in views.py to always display all rides when rids are searched for
Fixex floating footer issue in UI

Darius-Ndubi committed:
[Chore #158712504]

Binds the action to creeate rides to class users
Binds the action to edit rides to class users
Binds the action to request ride to class users

Darius-Ndubi committed:
Adds sign up validation and signing up
Adds signin validation and user registration
Adds Logged in user ability to create ride and un logged in user denial to perform action
Restricts data delition to users whom have created the ride
Adds visitor view to rides offerd
Adds visitor view to rides details
Restricts booking of a ride to registerd and logged in users
Adds the fuctionality of logged in user to make a ride request

Darius-Ndubi committed:
Adds responses to rides
Adds user signout
Adds user ride edit
